File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* twoSum(int* num, int n, int target, int* returnSize) {
    
    static int res[2];
    *returnSize=2;
    int left=0,right=n-1;
     
     while(left<right){
        if(num[left]+num[right]==target){
            res[0]=left+1;
            res[1]=right+1;
            break;
        }
        else if(num[left]+num[right]>target){
            right--;
        }
        else{
            left++;
        }
     }

     return res;

}

File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        int left=0,right=nums.length-1;

        while(left<right){
            if(nums[left]+nums[right]==target){
                return new int[]{left+1,right+1};
            }
            else if(nums[left]+nums[right]<target){
                left++;
            }
            else{
                right--;
            }
        }

        return nums;//dummy
    }
}


/*
HashMap<Integer,Integer> map=new HashMap<>();

        for(int i=0;i<nums.length;i++){
            int num1=nums[i];
            int num2=target-num1;

            if(map.containsKey(num2)==true){
                int j=map.get(num2);
                return new int[]{j+1,i+1};
            }
            map.put(num1,i);//adding in map

        }
        return nums;//dummy
*/